internal/daemon/cluster/handlers/worker_service.go

```diff
@@ -306,7 +306,10 @@ func (ws *workerServiceServer) LookupSession(ctx context.Context, req *pbs.Looku
 		if err != nil {
 			event.WriteError(ctx, op, err, event.WithInfoMsg("error looking up worker", "worker_id", req.WorkerId))
 			return &pbs.LookupSessionResponse{}, status.Errorf(codes.Internal, "Error looking up worker: %v", err)
-
+		}
+		if w == nil {
+			event.WriteError(ctx, op, err, event.WithInfoMsg("error looking up worker", "worker_id", req.WorkerId))
+			return &pbs.LookupSessionResponse{}, status.Errorf(codes.Internal, "Worker not found")
```

Comment:
Ugh this is such a dangerous API, to return `nil, nil` when the worker could not be found. I know this isn't the only place we do this but if this bug is any indicator it's not something we should continue doing.